spccore/multipart_upload.py

```diff
@@ -3,14 +3,34 @@
 
 from .constants import *
 from .internal.fileutils import *
+from .internal.pool_provider import *
 from .utils import *
 
 
 def multipart_upload_file(client,
                           file_path: str,
                           content_type: str,
                           *,
-                          storage_location_id: int = SYNAPSE_DEFAULT_STORAGE_LOCATION_ID):
+                          storage_location_id: int = SYNAPSE_DEFAULT_STORAGE_LOCATION_ID,
+                          generate_preview: bool = False,
+                          force_restart: bool = False,
+                          use_multiple_threads: bool = True) -> int:
```

Comment:
insead of use_multiple_threads, takes in pool provider as arg. and the higher level client is responsible for tracking which one to use